### ISSUE TITLE
thread 'tokio-runtime-worker' panicked at 'called `Option::unwrap()`

### ISSUE THREAD

**Issue by mikerudge** [prisma/prisma#10606]
### Bug description

When adding many to many relationships, I get this error when running Prisma client. This happens even when running `prisma studio`

```
thread 'tokio-runtime-worker' panicked at 'called `Option::unwrap()` on a `None` value', query-engine/connectors/mongodb-query-connector/src/root_queries/read.rs:106:74
```

### How to reproduce

1. Create a many to many models
2. Run `prisma studio`
3. Go to model

### Prisma information

```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["mongodb"]
  binaryTargets   = ["native", "rhel-openssl-1.0.x"]
}
```

```
model Organisation {
  id             String          @id @default(dbgenerated()) @map("_id") @db.ObjectId
  name           String
  activityGroups ActivityGroup[]
}

model ActivityGroup {
  id            String         @id @default(dbgenerated()) @map("_id") @db.ObjectId
  name          String
  organisations Organisation[]
}
```

```
prisma.organisation.findMany({
  take: 100,
  skip: 0,
  select: {
    id: true,
    name: true,
    activityGroups: true,
  }
})
```

### Environment & setup

- OS: Mac OS
- Database: MongoDB
- Node.js version: 14.17.6

### Prisma Version

```
3.2.1
```

**Comment by janpio** [comment]
Can you please try this with a more recent version of Prisma to make sure this did not already get fixed in the meantime before we start to investigate? Thanks.

**Comment by mikerudge** [comment]
> Can you please try this with a more recent version of Prisma to make sure this did not already get fixed in the meantime before we start to investigate? Thanks.

Oh my bad. I actually encountered this on the latest version but then tried previous versions back to 3.2 to see if it was a recent thing.

**Comment by HannesArni** [comment]
I encountered the same error at `query-engine/core/src/query_document/parser.rs:250:87`

In my case I was trying to create a record with a 22 decimal integer for a column with type `Int`. Changing that column and the value to a string fixed the issue for me. 
- Prisma version: `3.7.0`
- Database: Postgres
- Node version: `16.11.1`
- Os: Mac OS Big Sur

**Comment by janpio** [comment]
Did you also experience this error in Prisma Studio @HannesArni? Can you maybe create your own, separate issue? Thanks.

**Comment by janpio** [comment]
@mikerudge This is not valid syntax for MongoDB M:n relations, which we unfortunately do not validate and tell you about. I created #11481 to fix this. 

Here is the actual M:N syntax for MongoDB right now: https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#mongodb

**Comment by janpio** [comment]
This will now not pass validation any more with the next release:

```
C:\Users\Jan\Documents\throwaway\issue-repro-9072>npx prisma validate
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Error: Schema parsing
error: Error validating: Implicit many-to-many relations are not supported on MongoDB. https://pris.ly/d/many-to-many-relations
  -->  schema.prisma:24
   | 
23 |   name          String
24 |   organisations Organisation[]
25 | }
   | 
error: Error validating: Implicit many-to-many relations are not supported on MongoDB. https://pris.ly/d/many-to-many-relations
  -->  schema.prisma:18
   | 
17 |   name           String
18 |   activityGroups ActivityGroup[]
19 | }
   | 

Validation Error Count: 2
```
